File: OTCDerivativesCalculatorModule/Project_Cpp/lib_static/FpmlSerialized/GenClass/fpml-eq-shared-5-4/PrincipalExchangeAmount.cpp
// PrincipalExchangeAmount.cpp 
#include "PrincipalExchangeAmount.hpp"
#ifdef ConsolePrint
    #include <iostream>
#endif 

namespace FpmlSerialized {

PrincipalExchangeAmount::PrincipalExchangeAmount(TiXmlNode* xmlNode)
: ISerialized(xmlNode)
{
    #ifdef ConsolePrint
        std::string initialtap_ = FileManager::instance().tap_;
        FileManager::instance().tap_.append("   ");
    #endif 
   //amountRelativeToNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* amountRelativeToNode = xmlNode->FirstChildElement("amountRelativeTo");

   if(amountRelativeToNode){amountRelativeToIsNull_ = false;}
   else{amountRelativeToIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- amountRelativeToNode , address : " << amountRelativeToNode << std::endl;
   #endif
   if(amountRelativeToNode)
   {
      if(amountRelativeToNode->Attribute("href") || amountRelativeToNode->Attribute("id"))
      {
          if(amountRelativeToNode->Attribute("id"))
          {
             amountRelativeToIDRef_ = amountRelativeToNode->Attribute("id");
             amountRelativeTo_ = boost::shared_ptr<AmountReference>(new AmountReference(amountRelativeToNode));
             amountRelativeTo_->setID(amountRelativeToIDRef_);
             IDManager::instance().setID(amountRelativeToIDRef_,amountRelativeTo_);
          }
          else if(amountRelativeToNode->Attribute("href")) { amountRelativeToIDRef_ = amountRelativeToNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { amountRelativeTo_ = boost::shared_ptr<AmountReference>(new AmountReference(amountRelativeToNode));}
   }

   //determinationMethodNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* determinationMethodNode = xmlNode->FirstChildElement("determinationMethod");

   if(determinationMethodNode){determinationMethodIsNull_ = false;}
   else{determinationMethodIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- determinationMethodNode , address : " << determinationMethodNode << std::endl;
   #endif
   if(determinationMethodNode)
   {
      if(determinationMethodNode->Attribute("href") || determinationMethodNode->Attribute("id"))
      {
          if(determinationMethodNode->Attribute("id"))
          {
             determinationMethodIDRef_ = determinationMethodNode->Attribute("id");
             determinationMethod_ = boost::shared_ptr<DeterminationMethod>(new DeterminationMethod(determinationMethodNode));
             determinationMethod_->setID(determinationMethodIDRef_);
             IDManager::instance().setID(determinationMethodIDRef_,determinationMethod_);
          }
          else if(determinationMethodNode->Attribute("href")) { determinationMethodIDRef_ = determinationMethodNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { determinationMethod_ = boost::shared_ptr<DeterminationMethod>(new DeterminationMethod(determinationMethodNode));}
   }

   //principalAmountNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* principalAmountNode = xmlNode->FirstChildElement("principalAmount");

   if(principalAmountNode){principalAmountIsNull_ = false;}
   else{principalAmountIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- principalAmountNode , address : " << principalAmountNode << std::endl;
   #endif
   if(principalAmountNode)
   {
      if(principalAmountNode->Attribute("href") || principalAmountNode->Attribute("id"))
      {
          if(principalAmountNode->Attribute("id"))
          {
             principalAmountIDRef_ = principalAmountNode->Attribute("id");
             principalAmount_ = boost::shared_ptr<NonNegativeMoney>(new NonNegativeMoney(principalAmountNode));
             principalAmount_->setID(principalAmountIDRef_);
             IDManager::instance().setID(principalAmountIDRef_,principalAmount_);
          }
          else if(principalAmountNode->Attribute("href")) { principalAmountIDRef_ = principalAmountNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { principalAmount_ = boost::shared_ptr<NonNegativeMoney>(new NonNegativeMoney(principalAmountNode));}
   }

    #ifdef ConsolePrint
        FileManager::instance().tap_ = initialtap_;
    #endif 
}
boost::shared_ptr<AmountReference> PrincipalExchangeAmount::getAmountRelativeTo()
{
   if(!this->amountRelativeToIsNull_){
        if(amountRelativeToIDRef_ != ""){
             return boost::shared_static_cast<AmountReference>(IDManager::instance().getID(amountRelativeToIDRef_));
        }else{
             return this->amountRelativeTo_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<AmountReference>();
   }
}
boost::shared_ptr<DeterminationMethod> PrincipalExchangeAmount::getDeterminationMethod()
{
   if(!this->determinationMethodIsNull_){
        if(determinationMethodIDRef_ != ""){
             return boost::shared_static_cast<DeterminationMethod>(IDManager::instance().getID(determinationMethodIDRef_));
        }else{
             return this->determinationMethod_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<DeterminationMethod>();
   }
}
boost::shared_ptr<NonNegativeMoney> PrincipalExchangeAmount::getPrincipalAmount()
{
   if(!this->principalAmountIsNull_){
        if(principalAmountIDRef_ != ""){
             return boost::shared_static_cast<NonNegativeMoney>(IDManager::instance().getID(principalAmountIDRef_));
        }else{
             return this->principalAmount_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<NonNegativeMoney>();
   }
}
}

File: OTCDerivativesCalculatorModule/Project_CSharp/XmlBuilder/XmlSerializedClass/fpml-eq-shared-5-4/PrincipalExchangeAmount.cs
using System;
using System.Text;
using System.Collections.Generic;
using System.Xml;

namespace FpmlSerializedCSharp
{
    public class PrincipalExchangeAmount : ISerialized
    {
        public PrincipalExchangeAmount(XmlNode xmlNode)
        {
            XmlNode amountRelativeToNode = xmlNode.SelectSingleNode("amountRelativeTo");
            
            if (amountRelativeToNode != null)
            {
                if (amountRelativeToNode.Attributes["href"] != null || amountRelativeToNode.Attributes["id"] != null) 
                {
                    if (amountRelativeToNode.Attributes["id"] != null) 
                    {
                        amountRelativeToIDRef_ = amountRelativeToNode.Attributes["id"].Value;
                        AmountReference ob = new AmountReference(amountRelativeToNode);
                        IDManager.SetID(amountRelativeToIDRef_, ob);
                    }
                    else if (amountRelativeToNode.Attributes["href"] != null)
                    {
                        amountRelativeToIDRef_ = amountRelativeToNode.Attributes["href"].Value;
                    }
                    else
                    {
                        amountRelativeTo_ = new AmountReference(amountRelativeToNode);
                    }
                }
                else
                {
                    amountRelativeTo_ = new AmountReference(amountRelativeToNode);
                }
            }
            
        
            XmlNode determinationMethodNode = xmlNode.SelectSingleNode("determinationMethod");
            
            if (determinationMethodNode != null)
            {
                if (determinationMethodNode.Attributes["href"] != null || determinationMethodNode.Attributes["id"] != null) 
                {
                    if (determinationMethodNode.Attributes["id"] != null) 
                    {
                        determinationMethodIDRef_ = determinationMethodNode.Attributes["id"].Value;
                        DeterminationMethod ob = new DeterminationMethod(determinationMethodNode);
                        IDManager.SetID(determinationMethodIDRef_, ob);
                    }
                    else if (determinationMethodNode.Attributes["href"] != null)
                    {
                        determinationMethodIDRef_ = determinationMethodNode.Attributes["href"].Value;
                    }
                    else
                    {
                        determinationMethod_ = new DeterminationMethod(determinationMethodNode);
                    }
                }
                else
                {
                    determinationMethod_ = new DeterminationMethod(determinationMethodNode);
                }
            }
            
        
            XmlNode principalAmountNode = xmlNode.SelectSingleNode("principalAmount");
            
            if (principalAmountNode != null)
            {
                if (principalAmountNode.Attributes["href"] != null || principalAmountNode.Attributes["id"] != null) 
                {
                    if (principalAmountNode.Attributes["id"] != null) 
                    {
                        principalAmountIDRef_ = principalAmountNode.Attributes["id"].Value;
                        NonNegativeMoney ob = new NonNegativeMoney(principalAmountNode);
                        IDManager.SetID(principalAmountIDRef_, ob);
                    }
                    else if (principalAmountNode.Attributes["href"] != null)
                    {
                        principalAmountIDRef_ = principalAmountNode.Attributes["href"].Value;
                    }
                    else
                    {
                        principalAmount_ = new NonNegativeMoney(principalAmountNode);
                    }
                }
                else
                {
                    principalAmount_ = new NonNegativeMoney(principalAmountNode);
                }
            }
            
        
        }
        
    
        #region AmountRelativeTo_
        private AmountReference amountRelativeTo_;
        public AmountReference AmountRelativeTo_
        {
            get
            {
                if (this.amountRelativeTo_ != null)
                {
                    return this.amountRelativeTo_; 
                }
                else if (this.amountRelativeToIDRef_ != null)
                {
                    amountRelativeTo_ = IDManager.getID(amountRelativeToIDRef_) as AmountReference;
                    return this.amountRelativeTo_; 
                }
                else
                {
                      return this.amountRelativeTo_; 
                }
            }
            set
            {
                if (this.amountRelativeTo_ != value)
                {
                    this.amountRelativeTo_ = value;
                }
            }
        }
        #endregion
        
        public string amountRelativeToIDRef_ { get; set; }
        #region DeterminationMethod_
        private DeterminationMethod determinationMethod_;
        public DeterminationMethod DeterminationMethod_
        {
            get
            {
                if (this.determinationMethod_ != null)
                {
                    return this.determinationMethod_; 
                }
                else if (this.determinationMethodIDRef_ != null)
                {
                    determinationMethod_ = IDManager.getID(determinationMethodIDRef_) as DeterminationMethod;
                    return this.determinationMethod_; 
                }
                else
                {
                      return this.determinationMethod_; 
                }
            }
            set
            {
                if (this.determinationMethod_ != value)
                {
                    this.determinationMethod_ = value;
                }
            }
        }
        #endregion
        
        public string determinationMethodIDRef_ { get; set; }
        #region PrincipalAmount_
        private NonNegativeMoney principalAmount_;
        public NonNegativeMoney PrincipalAmount_
        {
            get
            {
                if (this.principalAmount_ != null)
                {
                    return this.principalAmount_; 
                }
                else if (this.principalAmountIDRef_ != null)
                {
                    principalAmount_ = IDManager.getID(principalAmountIDRef_) as NonNegativeMoney;
                    return this.principalAmount_; 
                }
                else
                {
                      return this.principalAmount_; 
                }
            }
            set
            {
                if (this.principalAmount_ != value)
                {
                    this.principalAmount_ = value;
                }
            }
        }
        #endregion
        
        public string principalAmountIDRef_ { get; set; }
        public string choiceStr_0;
        
    
        
    
    }
    
}

